tests/functional/aws-node-sdk/test/object/getObj.js

```diff
@@ -0,0 +1,216 @@
+            */
+            const UrlTest = [String.fromCharCode(1), '_url_test'].join('');
+            const s3 = bucketUtil.s3;
+            /**
+            * Put 1200 objects with / as delimiter
+            */
+            for (let i = 0; i !== 1200; ++i) {
+                const Prefix = validPrefix[0];
+                const Key = [Prefix, i.toString(), '_key_', UrlTest].join('');
+                const objects = [
+                    { Bucket, Key, ACL: 'public-read' },
+                ];
+                Promise
+                .mapSeries(objects, param => s3.putObjectAsync(param))
+                .then(() => s3.putObjectAclAsync(objects));
+            }
+
+            /**
+            * Put 250 objects with | as delimiter
+            */
+            for (let i = 0; i !== 250; ++i) {
+                const Prefix = validPrefix[1];
+                const Key = [Prefix, i.toString(), '_key_', UrlTest].join();
+                const objects = [
+                    { Bucket, Key, ACL: 'public-read' },
+                ];
+                Promise
+                .mapSeries(objects, param => s3.putObjectAsync(param))
+                .then(() => s3.putObjectAclAsync(objects));
+            }
+        };
+
+        /**
+        * Create bucket if is not created
+        * If the bucket is already create, we must empty it
+        */
+        bucketUtil.createOne(bucketName)
+        .then(() => {
+            generateDataSet();
+            done();
+        }).catch(() => {
+            bucketUtil.empty(bucketName).then(() => {
+                generateDataSet();
+                done();
+            });
+        });
+    });
+
+    after(done => {
+        const bucketUtil = new BucketUtility('default');
+
+        bucketUtil.empty(bucketName).then(() => {
+            done();
+            bucketUtil.deleteOne(bucketName);
+        }).catch(done);
+    });
+
+    const matrix = new testing.matrix.TestMatrix(params);
+
+    matrix.generate(['auth'], matrix => {
+        const bucketUtil = new BucketUtility('default', matrix.params.auth);
+
+        matrix.generate(['Delimiter', 'Prefix', 'MaxKeys', 'EncodingType',
+        'Bucket'], (matrix, done) => {
+            /**
+            * usual test
+            */
+            delete matrix.params.auth;
+            bucketUtil.s3.listObjects(matrix.params, (err, data) => {
+                const MaxKeys = matrix.params.MaxKeys;
+                const maxNumberOfKeys = Math.min(1200, MaxKeys);
+
+                const prefix = matrix.params.Prefix;
+                const delimiter = matrix.params.Delimiter;
+
+                const isPrefixMatch = (delimiter && prefix)
+                ? prefix.indexOf(delimiter) !== -1
+                : false;
+                const isGoodPrefix = validPrefix.indexOf(prefix) !== -1
+                || prefix === undefined;
+                if (isGoodPrefix === false) {
+                    assert.equal(err === null, true);
```

Comment:
If we want meaningful errors, it's `assert.strictEqual(err, null)` that we really want to see.